### PR TITLE
[http/client] expose basic tls config

### PR DESCRIPTION
**Pull request by kcking**
exposing these tls options enabled me to get https _with_ server verification working like so

```rs
    let mut client = EspHttpClient::new(&EspHttpClientConfiguration {
        use_global_ca_store: true,

        crt_bundle_attach: Some(esp_idf_sys::esp_crt_bundle_attach),

        ..Default::default()
    })?;
```

### CODE REVIEW

**Comment by ivmarkov** [comment]
The  `crt_bundle_attach` API you are using was introduced in the not-yet-released ESP-IDF V4.4+, while the minimum supported ESP-IDF version (and the official latest release) is still V4.3.X.

As you can check, it is not available here (V4.3.X): https://github.com/espressif/esp-idf/blob/release/v4.3/components/esp_http_client/include/esp_http_client.h#L108

**No prob with that**, but you need to conditionalize this code, as well as the `crt_bundle_attach` new member of struct `Configuration` to only compile with ESP-IDF versions later than 4.3.X.

The easiest way to do this to use `#[cfg(not(esp_idf_version = "4.3"))]` above your field, and also above the assignment statement further down. That would work (even though Rust does not support `<` and `>`), because versions earlier than 4.3.X are not supported as a compilation target by these crates anyway.

**Comment by kcking** [comment]
Good catch :) just made that change

Also I should note that this PR https://github.com/esp-rs/esp-idf-sys/pull/57 has the added import in the `sys` crate. Looks like that PR suffers from the same version compatibility issue, I'll fix it as well.

File: src/http/client.rs [hunk 1]
```diff
@@ -4,12 +4,13 @@ use alloc::collections::BTreeMap;
 use alloc::string::String;
 use alloc::string::ToString;
 
-use ::log::*;
+use log::*;
```
Comment (ivmarkov) [inline src/http/client.rs:7]: Why?

**Review by ivmarkov — CHANGES_REQUESTED** [review]
Only one superminor question for a changed import statement. The rest is perfect.

File: src/http/client.rs (hunk 1 above)
Comment (kcking) [reply src/http/client.rs]: Sorry, my auto cargo fmt in vscode must have changed that. Just fixed it!

**Comment by ivmarkov** [comment]
Thanks!

**Pull request merged**